internal/src/scheduling.ts

```diff
@@ -42,14 +22,43 @@ function clearInterval(tid?: number) {
   server.scheduler.cancelTask(tid);
 }
 
+/**
+ * Executes a function in server thread after a delay.
+ * @param handler Function to execute.
+ * @param delay Delay in milliseconds.
+ * @returns Task id.
+ */
 function setTimeout(handler: () => void, delay: number) {
-  const runnable = getRunnable(handler);
-  const runTaskLater = getSpigotMethod('runTaskLater');
-  const task = runTaskLater(__plugin, runnable, delay / 50) as BukkitTask;
-  return task.getTaskId();
+  return (__plugin as any).scheduleOnce(handler, Math.floor(delay / MILLIS_PER_TICK)) as number;
 }
 
+// Override JS standard library timers
 global.setInterval = setInterval as any;
 global.setTimeout = setTimeout as any;
 global.clearInterval = clearInterval;
 global.clearTimeout = clearInterval;
+
+
+// See global.wait for docs
+function wait(delay: number = 0, unit: TimeUnit = TimeUnit.TICKS): Promise<void> {
+  let ms: number;
+  switch (unit) {
+    case TimeUnit.MILLIS:
+      ms = delay;
+      break;
+    case TimeUnit.TICKS:
+      ms = delay * MILLIS_PER_TICK;
+      break;
+    case TimeUnit.SECONDS:
+      ms = delay * 1000;
+      break;
+    case TimeUnit.MINUTES:
+      ms = delay * 1000 * 60
+      break;
+    default:
+      throw 'unknown time unit';
```

Comment:
Throwing strings should probably be avoided as they can mess up error handling code
```suggestion
      throw Error('unknown time unit');
```